packages/docs/pages/usage/typescript.mdx

```diff
@@ -0,0 +1,49 @@
+
+
+# TypeScript
+
+## Configuration
+Set `moduleResolution` in the `compilerOptions` in your `tsconfig.json` to be either `Node16` or `NodeNext`.
+
+When importing types, always refer to the standard orama import:
+
+```ts copy
+import type { Language } from '@orama/orama'
+```
+
+## Usage
+
+Orama is written in TypeScript and provides type definitions for all of its public APIs.
+This means the results of all functions are typed, and the types are **inferred from the schema provided**.
+
+```typescript copy
+const db = await create({
+  schema: {
+    title: 'string',
+    plot: 'string',
+  }
+})
+await insert(db, {
+  title: 'The prestige',
+  plot: 'Two friends and fellow magicians become bitter enemies after a sudden tragedy. As they devote themselves to this rivalry, they make sacrifices that bring them fame but with terrible consequences.'
+})
+const s = await search(db, { term: 'prestige' })
+
+const firstTitle: string = s.hits[0].document.title // Inferred by the schema provided
+```
+
+Anyway, because Orama schema defines only searchable fields, you may want to add some extra fields to your documents.
+Extra fields are not indexed, so they are not searchable, but they are still available in the results.
+The type of these fields is not inferred, so it's `any`.
+
+```typescript copy
+await insert(db, {
+  title: 'Big Fish',
+  director: 'Tim Burton',
+  plot: 'Will Bloom returns home to care for his dying father, who had a penchant for telling unbelievable stories. After he passes away, Will tries to find out if his tales were really true.',
+  year: 2004,
+  isFavorite: true
+})
+const s = await search(db, { term: 'fish' })
+const year: any = s.hits[0].document.year // not inferred, so it's any
```

Comment:
It's not inferred but you can make it type-safe by passing the interface with that field, like: `search<WithYearField>`

Reply:
good point. I'm changing it!

Reply:
The tests that you have on Orama don't test for types, you can literally break the types and the tests will not break, they only break when the assert is false (at least that was the behavior I had when I was changing the Orama tests).

But I can add it by the end of the day.

Reply:
Yeah, till now, we don't have that kind of test. But with this PR, type types are guaranteed. So, a little test on that would be amazing!
At least, we document this capability in the test suite also.

Reply:
In this case, we can use: https://github.com/SamVerschueren/tsd

Is the same lib used by https://github.com/sindresorhus/type-fest/blob/main/test-d/abstract-class.ts

Reply:
But probably this will require a new project and command to test only the types, separated from the default tap tests.

Reply:
`tsd` seems to do the job greatly.
I'll go for just a new command run after the test suite. it should work.